tensorpipe/test/channel/channel_test_cuda.cc

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <cuda_runtime.h>
+
+#include <tensorpipe/test/channel/channel_test.h>
+#include <tensorpipe/test/channel/kernel.cuh>
+
+using namespace tensorpipe;
+using namespace tensorpipe::channel;
+
+class ReceiverWaitsForStartEventTest
+    : public ClientServerChannelTestCase<tensorpipe::CudaBuffer> {
```

Comment:
Also probably not needed?
```suggestion
    : public ClientServerChannelTestCase<CudaBuffer> {
```